src/components/Login/Login.tsx

```diff
@@ -0,0 +1,163 @@
+import { getCsrfToken, signIn } from "next-auth/react";
+import { useForm } from "react-hook-form";
+
+import { useState, useRef, useEffect } from "react";
+
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+  DialogFooter,
+  Switch,
+  Input,
+  Label,
+  Button,
+} from "~/components/ui";
+
+interface FormData {
+  email: string;
+  password: string;
+  type: string;
+}
+
+import { auth } from "~/pages/api/auth/[...nextauth]";
+import { sendPasswordResetEmail } from "firebase/auth";
+
+const Login = () => {
+  const [submitType, setSubmitType] = useState("sign-in");
+  const [formError, setFormError] = useState(" ");
+  const [wrongPassCount, setWrongPassCount] = useState(0);
+
+  const csrfInput = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    (async () => {
+      const csrfToken = await getCsrfToken();
+      if (csrfInput.current) {
+        csrfInput.current.value = csrfToken as string;
+      }
+    })();
+  });
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<FormData>();
+
+  const onSubmit = async (data: FormData) => {
+    const signInRes = await signIn(submitType, { ...data, redirect: false });
+    if (signInRes && signInRes.error) {
+      if (signInRes.error === "Too many requests...") {
+        setWrongPassCount(wrongPassCount + 1);
+      }
+
+      setFormError(signInRes.error);
+    }
+  };
+  return (
+    <Dialog>
+      <DialogTrigger>Sign in / Sign Up</DialogTrigger>
+      <DialogContent>
+        <DialogHeader>
+          <DialogTitle>
+            {submitType === "sign-in" ? "Sign In" : "Sign Up"}
+          </DialogTitle>
+          <DialogDescription>
+            {submitType === "sign-in"
+              ? "Login into existing account"
+              : "Create a new account"}
+          </DialogDescription>
+        </DialogHeader>
+        <form
+          className="flex flex-col items-center justify-center gap-8"
+          method="post"
+          action="/api/auth/callback/credentials"
```

Comment:
why you've left it? you are doing signin by hands anyway and useForm prevents form event